src/actions/ratesActions.js

```diff
@@ -51,35 +57,49 @@ export const setRatesAction = (newRates: Rates) => {
   };
 };
 
-export const fetchAccountAssetsRatesAction = () => {
+export const fetchAssetsRatesAction = () => {
   return async (dispatch: Dispatch, getState: GetState) => {
-    const accountAssets = accountAssetsCompatSelector(getState());
-    const rates = await getExchangeRates(accountAssets);
-    dispatch(setRatesAction(rates));
-  };
-};
+    const assetsBalances = assetsBalancesSelector(getState());
+    const supportedAssets = supportedAssetsSelector(getState());
 
-export const fetchAllAccountsAssetsRatesAction = () => {
-  return async (dispatch: Dispatch, getState: GetState) => {
-    const allAccountsAssets = assetsCompatSelector(getState());
+    const allAccountsCrossChainAssets = Object.keys(assetsBalances).reduce((combinedAssets, accountId) => {
+      const accountAssetsBalances = assetsBalances[accountId] ?? {};
+
+      const accountAssets = Object.keys(accountAssetsBalances).reduce((assets, chain) => {
+        const chainSupportedAssets = supportedAssets[chain] ?? [];
+        const walletAssets = accountAssetsBalances[chain]?.wallet ?? {};
+        const mapped = mapWalletAssetsBalancesIntoAssets(walletAssets, chainSupportedAssets);
+        // $FlowFixMe
+        return { ...assets, ...mapped };
+      }, {});
 
-    // check if not empty just in case
-    if (isEmpty(allAccountsAssets)) {
-      reportErrorLog('fetchAllAccountsAssetsRatesAction failed: empty all account assets', { allAccountsAssets });
+      return { ...combinedAssets, ...accountAssets };
+    }, {});
+
+    if (isEmpty(allAccountsCrossChainAssets)) {
+      reportErrorLog('fetchAssetsRatesAction failed: allAccountsCrossChainAssets is empty');
       return;
     }
 
-    const allAssets = mergeAccountAssets(allAccountsAssets);
-    const rates = await getExchangeRates(allAssets);
+    const rates = await getExchangeRates(allAccountsCrossChainAssets);
     dispatch(setRatesAction(rates));
   };
 };
 
-export const fetchSingleAssetRatesAction = (assetCode: string) => {
+export const fetchSingleChainAssetRatesAction = (
```

Comment:
That function accepts chain argument, but `getExchangeRates` is actually querying mainnet. Need to change underlying coingecko queries.

Reply:
Will fix rates per chain in separate PR.